indigo/analysis/toc/base.py

```diff
@@ -245,10 +257,7 @@ def commenceable_items(self, toc):
         """
         def process(item):
             if item.component == 'main':
-                if item.children:
-                    for kid in item.children:
-                        process(kid)
-                elif item.type not in self.non_commenceable_types and item.num:
+                if item.type not in self.non_commenceable_types and item.num:
```

Comment:
Just noting that this no longer recurses into children, so it's returning a tree structure rather than a list.